_blogposts/compiler/overview-of-new_encoding.mdx

```diff
@@ -0,0 +1,245 @@
+---
+author: hongbo
+date: "2020-06-10 18:00" 
+previewImg:
+category: compiler
+title: Overview of new encoding in BuckleScript
+description: |
+  Highlights of our newest changes to the internal representation 
+  and how they will benefit our users.
+---
+
+
+---
+title: An overview of new data representation in BuckleScript
+---
+
+In the next version of BuckleScript, we will make several major changes to tweak the data representation for various data types, 
+making them more idiomatic and debugger friendly. 
+
+Note: since V8 or other JavaScript engines are tweaked to make idiomatic JS code run fast, these changes also results in faster running code. 
+
+Another property, for a compiled language like BuckleScript, 
+is that we can reason about [IC](https://en.wikipedia.org/wiki/Inline_caching) friendliness by just looking at the type definitions locally; 
+this is very helpful for advanced users to write performance predictable JS code. 
+If you are unfamiliar with how IC works in general, feel free to skip the section about IC.
+
+Note: this article is quite dense, so  we will skip the old encoding.
+
+## Record (stable)
```

Comment:
`[@bs.as "0"]` is new, it is planned as overview, since this post covers most data representations. exception is not mentioned since it is not a common data type

Reply:
okay, i was trying `[@bs.as "0"]` in the `7.4.0-dev.1` based online playground and it compiled to something, so I thought it was already there